lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -420,39 +449,27 @@ def shutdown
     end
 
     def write(chunk)
-      # Group the entries since we have to make one call per tag.
-      grouped_entries = {}
-      chunk.msgpack_each do |tag, *arr|
-        sanitized_tag = sanitize_tag(tag)
-        if sanitized_tag.nil?
-          @log.warn "Dropping log entries with invalid tag: '#{tag}'. " \
-                    'A tag should be a string with utf8 characters.'
-          next
-        end
-        grouped_entries[sanitized_tag] ||= []
-        grouped_entries[sanitized_tag].push(arr)
-      end
+      grouped_entries = group_log_entries_by_tag_and_local_resource_id(chunk)
 
-      grouped_entries.each do |tag, arr|
+      grouped_entries.each do |(tag, local_resource_id), arr|
         entries = []
-        group_resource, group_common_labels =
-          determine_group_level_monitored_resource_and_labels(tag)
+        group_level_resource, group_level_common_labels =
```

Comment:
[Optional] I'm fine with keeping this as-is, but this renaming (`group_resource`->`group_level_resource`, `entry_resource`->`entry_level_resource`, etc) seems somewhat gratuitous to me.

Reply:
Ah, originally I renamed them because I added `tag_resource` and `tag_labels`. It's kind of ambiguous what that means. So i named them in an more explicit way showing that these are just regular monitored resource and metadata labels except they are at different levels.